tests/client/amo/components/TestAddonDetail.js

```diff
@@ -291,19 +298,27 @@ describe('AddonDetail', () => {
         ...fakeAddon,
         type: ADDON_TYPE_THEME,
       },
-      clientSupportsAddons: () => true,
+      getCompatibleClient: () => ({ compatible: true }),
     });
     const button = rootNode.querySelector('.AddonDetail-theme-header-label');
     assert.equal(button.disabled, false);
   });
 
+  it('disables install switch for unsupported clients', () => {
+    const rootNode = renderAsDOMNode({
+      getClientCompatibility: getClientCompatibilityFalse,
+    });
+    assert.isTrue(
+      rootNode.querySelector('.InstallButton-switch input').disabled);
+  });
+
   it('disables a theme preview for unsupported clients', () => {
     const rootNode = renderAsDOMNode({
       addon: {
         ...fakeAddon,
         type: ADDON_TYPE_THEME,
       },
-      clientSupportsAddons: () => false,
+      getClientCompatibility: getClientCompatibilityFalse,
```

Comment:
Does this test throw an error if you delete some required props from `<AddonCompatibilityError />` within the `AddonDetail` render function? If not, I'd like to see a test that makes sure `AddonDetail` is passing the right props to `AddonCompatibilityError`. You could use a stub component (passed via props) and make assertions about what props it received.